### DISCUSSION TITLE
"The "path" argument must be of type string. Received undefined" after upgrading to 5.21.1

### DISCUSSION THREAD

**Discussion by titusdecali** [prisma/prisma#25454] [Q&A] [answered]
### Bug description

Was going to give Optimize a test, but upgrading to 5.21.1 keeps causing the error below when running `prisma generate`

Can I use optimize on Prisma `5.17.0`?

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Error: 
The "path" argument must be of type string. Received undefined
```

I've reference the previous issues [6372](https://github.com/prisma/prisma/issues/6372) and [6770](https://github.com/prisma/prisma/issues/6770) already.

I'm using `5.21.1` for both `prisma` and `prisma/client` (without the carat) and I've tried wiping my `pnpm-lock.yaml` + `node_modules` folding and reinstalling. No dice. I didn't have this issue with `5.17.0`. I've also tried it on `node 18`, same result.

### How to reproduce

```
dependancies:
    "nuxt": "^3.13.2",
    "prisma": "5.21.1"
    "@prisma/extension-accelerate": "^1.2.1",
    "@prisma/extension-optimize": "^1.0.1",
devDependancies:
    "prisma": "5.21.1",
```
    

### Expected behavior

For the `prisma generate` command to not cause errors

### Prisma information

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["tracing", "multiSchema"]
}

datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_DATABASE_URL")
  schemas   = ["public"]
}

model account {
  id         String     @id @db.Uuid
  email      String?
  provider   String?
  avatar_url String?
  created_at DateTime   @default(now()) @db.Timestamptz(6)
  first_name String?
  full_name  String?
  last_name  String?
  updated_at DateTime   @default(now()) @db.Timestamptz(6)
  comment    comment[]
  feedback   feedback[]
  upvotes    upvote[]

  @@schema("public")
}

model comment {
  id             String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  content        String
  creator_name   String?
  creator_avatar String?
  created_at     DateTime @default(now()) @db.Timestamptz(6)
  creator_id     String?  @db.Uuid
  feedback_id    String   @db.Uuid
  updated_at     DateTime @default(now()) @db.Timestamptz(6)
  creator        account? @relation(fields: [creator_id], references: [id], onDelete: Restrict)
  feedback       feedback @relation(fields: [feedback_id], references: [id])

  @@schema("public")
}

model company {
  id                 String    @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  name               String
  logo               String?   @unique
  slug               String    @unique
  created_at         DateTime  @default(now()) @db.Timestamptz(6)
  owner_id           String    @db.Uuid
  updated_at         DateTime  @default(now()) @db.Timestamptz(6)
  stripe_customer_id String?
  subscription       String?
  subscription_id    String?   @unique
  members            member[]
  products           product[]

  @@schema("public")
}

model member {
  id         String   @db.Uuid
  role       role
  email      String?
  avatar_url String?
  company_id String   @db.Uuid
  created_at DateTime @default(now()) @db.Timestamptz(6)
  full_name  String?
  updated_at DateTime @default(now()) @db.Timestamptz(6)
  company    company  @relation(fields: [company_id], references: [id])

  @@id([id, company_id])
  @@schema("public")
}

model product {
  id            String       @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  logo          String?      @unique
  name          String
  description   String
  subdomain     String       @unique
  company_id    String       @db.Uuid
  created_at    DateTime     @default(now()) @db.Timestamptz(6)
  updated_at    DateTime     @default(now()) @db.Timestamptz(6)
  website       String?
  colors        Json?
  anon_settings Json         @default("{}")
  widget_key    String       @unique
  icon          String?      @unique
  roadmap_order Json?
  feedback      feedback[]
  company       company      @relation(fields: [company_id], references: [id])
  tags          tag[]        @relation("ProductTags")
  status_tags   status_tag[] @relation("StatusTags")

  @@schema("public")
}

model feedback {
  id                 String               @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  title              String
  description        String
  created_at         DateTime             @default(now()) @db.Timestamptz(6)
  creator_id         String               @db.Uuid
  product_id         String               @db.Uuid
  updated_at         DateTime             @default(now()) @db.Timestamptz(6)
  status             String               @default("notstarted")
  would_pay          Boolean?
  FeedbackStatusTags FeedbackStatusTags[]
  comments           comment[]
  creator            account              @relation(fields: [creator_id], references: [id], map: "feedback_creator_id_fkey")
  product            product              @relation(fields: [product_id], references: [id], map: "feedback_product_id_fkey")
  upvotes            upvote[]
  tags               tag[]                @relation("FeedbackTags")

  @@schema("public")
}

model upvote {
  id          String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  feedback_id String   @db.Uuid
  upvoter_id  String   @db.Uuid
  created_at  DateTime @default(now()) @db.Timestamptz(6)
  updated_at  DateTime @default(now()) @db.Timestamptz(6)
  feedback    feedback @relation(fields: [feedback_id], references: [id])
  upvoter     account  @relation(fields: [upvoter_id], references: [id])

  @@unique([feedback_id, upvoter_id])
  @@schema("public")
}

model tag {
  type     tagType    @default(standard)
  id       String     @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  name     String
  value    String     @unique
  public   Boolean
  order    Int?       @db.SmallInt
  feedback feedback[] @relation("FeedbackTags")
  products product[]  @relation("ProductTags")

  @@schema("public")
}

model status_tag {
  id                 String               @id(map: "status_pkey") @default(dbgenerated("gen_random_uuid()")) @db.Uuid
  name               String?
  value              String?
  public             Boolean?
  starting_order     Int?                 @db.SmallInt
  type               tagType?
  color              String?
  FeedbackStatusTags FeedbackStatusTags[]
  product            product[]            @relation("StatusTags")

  @@schema("public")
}

model stripe_customers {
  company_id         String   @id(map: "stripeCustomers_pkey")
  stripe_customer_id String?
  created_at         DateTime @default(now()) @db.Timestamptz(6)

  @@schema("public")
}

model FeedbackStatusTags {
  A          String     @db.Uuid
  B          String     @db.Uuid
  status_tag status_tag @relation(fields: [A], references: [id], onDelete: Cascade, map: "_FeedbacktatusTags_A_fkey")
  feedback   feedback   @relation(fields: [B], references: [id], onDelete: Cascade, map: "_FeedbacktatusTags_B_fkey")

  @@unique([A, B], map: "_FeedbacktatusTags_AB_unique")
  @@index([B], map: "_FeedbacktatusTags_B_index")
  @@map("_FeedbackStatusTags")
  @@schema("public")
}

enum tagType {
  standard
  custom

  @@schema("public")
}

enum role {
  owner
  admin
  member

  @@schema("public")
}

```

```ts
import { PrismaClient } from '@prisma/client'
import { withOptimize } from '@prisma/extension-optimize'
import { withAccelerate } from '@prisma/extension-accelerate'
const config = useRuntimeConfig()

const prisma = new PrismaClient()
  .$extends(withAccelerate())
  .$extends(
    withOptimize({ apiKey: config.private.PRISMA_OPTIMIZE_API_KEY as string })
  )

export default function usePrisma() {
  return { prisma }
}

```

### Environment & setup

- OS: Mac Ventura 13.6
- Database: Supabase
- Node.js version: `21.0.0` / `18`

### Prisma Version

```
dependancies:
    "prisma": "5.21.1"
    "@prisma/extension-accelerate": "^1.2.1",
    "@prisma/extension-optimize": "^1.0.1",
devDependancies:
    "prisma": "5.21.1",
```

**Comment by SevInf** [comment]
Hi @titusdecali.
Thank you for opening the issue. Yes, we need also the info we asking we asking for in the issue template for troubleshooting. Please, provide your schema.

**Comment by titusdecali** [comment]
@SevInf Schema added

**Comment by SevInf** [comment]
@titusdecali I can not reproduce your issue, `prisma generate` finishes succesfully for me. Could you share an isolated repository where we could see error happening?

**Comment by titusdecali** [comment]
@SevInf I've just set up a local starter project, and I am seeing the exact same errors. I've invited you to the repo.

Also, I noticed that when using `npx prisma` and then `npx prisma init` it is adding version `"5.11.0"`. and not `"5.21.1"`
If I run it with `"5.11.0"`, it actually works as expected

Perhaps this is the cause?

**Comment by SevInf** [comment]
@titusdecali I can not reproduce the issue just from your description. I've started a project, copied your schema and executed `generate` command and it work normally for me. Could you please share the repository you just started?

**Comment by titusdecali** [comment]
Yes, I've invited you directly, but it is here: https://github.com/titusdecali/nuxt3-prisma-bug-reproduction

**Comment by SevInf** [comment]
I've checked your repository out, run `npm install`, then `npx prisma generate` and it succefully completed to me. 
`npx prisma` resolving to 5.11.0 is suspicious, it does resolve to `5.21.1` to me. Could you try with `npx prisma@5.21.1` explicitly?

**Comment by titusdecali** [comment]
I am using `pnpm`, but `npm` causes the following issue
![SHOT 19 51 51@2x](https://github.com/user-attachments/assets/58bf9b48-02b3-4a1d-9c16-1c64e3aa9b19)

I've attached the log file it referenced: 
[2024-10-18T10_45_46_792Z-debug-0.log](https://github.com/user-attachments/files/17434238/2024-10-18T10_45_46_792Z-debug-0.log)

when manually setting the version in package.json, (as `npx prisma@5.21.1` didn't seem to do that for me) I get the same error as before:

![SHOT 19 54 42@2x](https://github.com/user-attachments/assets/8e82b95d-c46d-48ca-ba12-d528dd6bdaac)

**Reply by SevInf** [reply, marked as answer]
can your run `pnpm prisma generate` rather than `prisma generate`?

**Reply by sadiq-i-umar** [reply]
Thank you. In my own case, I was using prisma generate in the terminal instead of using npx prisma generate

**Comment by titusdecali** [comment]
Hmm... it's working, in both the demo project and my original one.
never had to use `pnpm` before `prisma generate` before now.

What caused the need to do that now with `5.21.1`?

**Comment by zipme** [comment]
Got the same issue, I am using bun, so running `bun prisma generate`  results in:
```
Error:
The "path" argument must be of type string. Received undefined
```

After downgrading to `5.17.0` it works again, it's broken on 5.18.0 and above

**Comment by coreh** [comment]
Hit this while attempting to upgrade to prisma 6.0.1, (using bun 1.1.38) and was able to finally resolve it after struggling with it for quite a bit.

👉 If you're on a monorepo, **make sure there aren't any packages that reference a prior version of either `"prisma"` or `"@prisma/client"` instead of the one you're upgrading to**, in my case I inadvertently had references to the prior version in a couple of my own `package.json` files.

👉 Beyond that, make sure you don't have **any stray, leftover packages deeply nested inside of child `node_modules` dirs that may also be referencing older versions**. Side effects in the package tree like this probably shouldn't happen normally, but my project somehow had quite a few of them. (Maybe caused a bug in prior versions of bun, prisma, or some other tool 🤷‍♂️) I believe this is why not everyone can consistently reproduce this.

To fix that, I had to do this:

```sh
rm -rf node_modules
rm -rf packages/*/node_modules
```

(Assuming `packages/` is where your monorepo packages are, adjust accordingly)

After doing this and reinstalling, `prisma generate` works normally under 6.0.1 for me.

**Comment by ruckerzerg** [comment]
Having the same issue with a newly created project and version 6.0.1.

**Reply by Ankishoop** [reply]
npm i @prisma/client@latest
 npm i --save-dev prisma@latest 

then migrate then generate works for me . (version6.0.1)

**Reply by ronitsharma03** [reply]
> Having the same issue with a newly created project and version 6.0.1.

Solution: 
Yeah for me too it worked after doing:

npm install --save-dev prisma@latest
npm install @prisma/client@latest

Then you can generate the client, then it will work as expected!

**Reply by ebharathi** [reply]
this worked thank god

**Comment by ronitsharma03** [comment]
Yeah for me too it worked after doing:

1. npm install --save-dev prisma@latest
2. npm install @prisma/client@latest

Actually it can be due to some compatibility issues between prisma's 5.21.1 version with the generated client version...
So updating them to the latest version solves the issue !

**Reply by andrewdownss** [reply]
follow this, and still getting the path error
"@prisma/client": "6.1.0",
"prisma": "6.1.0",

Error: 
The "path" argument must be of type string. Received undefined

EDIT: Nvm it worked

**Reply by ronitsharma03** [reply]
@andrewdownss For me it worked well as expected !
